internal/services/network/custom_ip_prefix_resource.go

```diff
@@ -0,0 +1,419 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-05-01/network"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/zones"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type commissionedAction string
+
+const (
+	commissionedActionProvision    commissionedAction = "Provision"
+	commissionedActionCommission   commissionedAction = "Commission"
+	commissionedActionDecommission commissionedAction = "Decommission"
+	commissionedActionDeprovision  commissionedAction = "Deprovision"
+)
+
+type commissionedState string
+
+const (
+	commissionedStateDeprovisioned    commissionedState = "Deprovisioned"
+	commissionedStateValidationFailed commissionedState = "ValidationFailed"
+)
+
+func resourceCustomIpPrefix() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCustomIpPrefixCreateUpdate,
+		Read:   resourceCustomIpPrefixRead,
+		Update: resourceCustomIpPrefixCreateUpdate,
```

Comment:
could we split this into separate create and update
```suggestion
		Create: resourceCustomIpPrefixCreate,
		Read:   resourceCustomIpPrefixRead,
		Update: resourceCustomIpPrefixUpdate,
```

Reply:
Fixed.